src/sagas/reown.js

```diff
@@ -0,0 +1,861 @@
+  
+  try {
+    const data = {
+      icon: get(params, 'proposer.metadata.icons[0]', null),
+      proposer: get(params, 'proposer.metadata.name', ''),
+      url: get(params, 'proposer.metadata.url', ''),
+      description: get(params, 'proposer.metadata.description', ''),
+      requiredNamespaces: get(params, 'requiredNamespaces', []),
+    };
+
+    let dispatch;
+    yield put((_dispatch) => {
+      dispatch = _dispatch;
+    });
+
+    const connectResponseTemplate = (accepted) => () => {
+      dispatch(hideGlobalModal());
+      if (accepted) {
+        dispatch({ type: types.REOWN_ACCEPT });
+      } else {
+        dispatch({ type: types.REOWN_REJECT });
+      }
+    };
+
+    // Show the modal
+    yield put(showGlobalModal(MODAL_TYPES.REOWN, {
+      type: ReownModalTypes.CONNECT,
+      data,
+      onAcceptAction: connectResponseTemplate(true),
+      onRejectAction: connectResponseTemplate(false),
+    }));
+
+    const { accepted } = yield race({
+      accepted: take(types.REOWN_ACCEPT),
+      rejected: take(types.REOWN_REJECT),
+    });
+
+    const { walletKit } = getGlobalReown();
+    if (!walletKit) {
+      throw new Error('WalletKit not initialized');
+    }
+
+    const networkSettings = yield select(getNetworkSettings);
+    log.debug('Network Settings: ', networkSettings);
+    if (accepted) {
+      const wallet = getGlobalWallet();
+      const firstAddress = yield call(() => wallet.getAddressAtIndex(0));
+
+      // User accepted the proposal
+      yield call([walletKit, walletKit.approveSession], {
+        id,
+        relayProtocol: params.relays[0].protocol,
+        namespaces: {
+          hathor: {
+            accounts: [`hathor:${networkSettings.network}:${firstAddress}`],
+            chains: [`hathor:${networkSettings.network}`],
+            events: AVAILABLE_EVENTS,
+            methods: values(AVAILABLE_METHODS),
+          },
+        },
+      });
+    } else {
+      // User rejected the proposal
+      yield call([walletKit, walletKit.rejectSession], {
+        id,
+        reason: {
+          code: ERROR_CODES.USER_REJECTED,
+          message: 'User rejected the session',
+        },
+      });
+    }
+
+    // Refresh the sessions list
+    yield call(refreshActiveSessions);
+  } catch (error) {
+    log.error('Error handling session proposal:', error);
+    yield put(onExceptionCaptured(error));
+  } finally {
+    // Make sure to close the modal even if there's an error
+    yield put(hideGlobalModal());
+  }
+}
+
+export function* onUriInputted(action) {
+  const { core, walletKit } = getGlobalReown();
+
+  if (!core || !walletKit) {
+    log.debug('Tried to get reown client in onUriInputted but core or walletKit is undefined.');
+    return;
+  }
+
+  const { payload } = action;
+
+  try {
+    yield call(core.pairing.pair, { uri: payload });
+  } catch (error) {
+    log.debug('Error pairing: ', error);
+    yield put(setWCConnectionFailed(true));
```

Comment:
Do you set it to `false` somewhere? Is it needed?

Reply:
Hey, I refactored the reown connection handling entirely, can you please check baa5cfa6645e4c77f90b0a03763869a9d5b1cb17?